src/streamlink/session.py

```diff
@@ -373,6 +374,10 @@ def set_option(self, key: str, value: Any) -> None:
               - ``float``
               - ``10.0``
               - Segment connect and read timeout
+            * - stream-segmented-duration
+              - ``float | None``
+              - ``None``
+              - Limit the playback duration of segmented streams, rounded to the nearest segment
```

Comment:
As an outsider, I still think this name is a bit confusing or hard to discover. Why are the other stream-segment options in present tense?

Reply:
- https://deploy-preview-5493--streamlink.netlify.app/cli#cmdoption-stream-segmented-duration
- https://deploy-preview-5493--streamlink.netlify.app/api/session#streamlink.session.Streamlink.set_option

We can change the description's imperative, to make it more consistent with the other descriptions. As you can see in the diff, this was copied from the old option.

I have already explained the reasoning behind the option/CLI-argument naming. If you have a better name, then suggest one. It can't be `stream-duration`, because it's limited to segmented streams. The `stream-segment-...` options are meant for logic that affects single segments in segmented streams.  The newly added `stream-segmented-...` options are for logic that affects segmented streams as a whole.

Reply:
Okay, no worries, I have no better suggestion that wouldn't at the same time give the impression that it works for all streams.